docs/architecture/adrs/007-orchestrator-telemetry.md

```diff
@@ -0,0 +1,74 @@
+# ADR 007: Orchestrator Telemetry
+
+The guardrails orchestrator uses OpenTelemetry to collect and export telemetry data (traces, metrics, and logs).
+
+The orchestrator needs to collect telemetry data for monitoring and observability. It also needs to be able to trace
+spans for incoming requests and across client requests to configured detectors, chunkers, and generation services and 
+aggregate details traces, metrics, and logs that can be monitored from a variety of observability backends.
+
+## Decision
+
+### OpenTelemetry and `tracing`
+
+The orchestrator and client services will make use of the OpenTelemetry SDK and the OTLP standard for consolidating
+and collecting telemetry data across services. The orchestrator will be responsible for collecting telemetry data
+throughout the lifetime of a request using the `tracing` crate, which is the de facto choice for logging and tracing
+for OpenTelemetry in Rust, and exporting it through the OTLP exporter if configured. The OTLP exporter will send 
+telemetry data to a gRPC or HTTP endpoint that can be configured to point to a running OTEL collector.
+Similarly, detectors should also be able to collect and export telemetry through OTLP to the same OTEL collector.
+From the OTEL collector, the telemetry data can then be exported to multiple backends. The OTEL collector and
+any observability backends can all be configured alongside the orchestrator and detectors in a deployment.
+
+### Instrumentation
+An incoming request to the orchestrator will initialize a new trace, therefore a trace-id and request should be in
+one-to-one correspondence. All important functions throughout the control flow of handling a request in the orchestrator 
+will be instrumented with the `#[tracing::instrument]` attribute macro. This will create and enter a span for each 
+function call and add it to the trace of the request. Here, important functions refers to any function that performs
+important business logic or that may incur significant latency, this includes all the handler functions for incoming
+and outgoing requests.
```

Comment:
Is it correct understanding that if we are adding a new function that qualifies for "significant processing", then we can simply enable tracing by adding that macro on top of the function ? 

Are there any other requirement for such a new function?

Reply:
Added a bit more clarification. "Significant" function is up to the discretion of the developer but yes adding tracing is as trivial as adding the instrument macro above the function definition